configs/3.10/post_extract.patch

```diff
@@ -0,0 +1,309 @@
+ 
+ 	return 0;
+@@ -338,14 +368,14 @@ int restore_sched_debug_procfs(void)
+ #endif
+ 
+ #ifdef CONFIG_SCHEDSTATS
+-extern struct seq_operations __orig_schedstat_sops;
++extern struct file_operations __orig_proc_schedstat_operations;
+ 
+ /* schedstat interface in proc */
+ int install_proc_schedstat(void)
+ {
+ 	remove_proc_entry("schedstat", NULL);
+ 
+-	if (!proc_create_seq("schedstat", 0444, NULL, &__mod_schedstat_sops))
++	if (!proc_create("schedstat", 0444, NULL, &__mod_proc_schedstat_operations))
+ 		return -ENOMEM;
+ 
+ 	return 0;
+@@ -355,7 +385,7 @@ int restore_proc_schedstat(void)
+ {
+ 	remove_proc_entry("schedstat", NULL);
+ 
+-	if (!proc_create_seq("schedstat", 0444, NULL, &__orig_schedstat_sops))
++	if (!proc_create("schedstat", 0444, NULL, &__orig_proc_schedstat_operations))
+ 		return -ENOMEM;
+ 
+ 	return 0;
+diff --git a/kernel/sched/mod/sched_rebuild.c b/kernel/sched/mod/sched_rebuild.c
+index 6e8a4c7..37427a8 100644
+--- a/kernel/sched/mod/sched_rebuild.c
++++ b/kernel/sched/mod/sched_rebuild.c
+@@ -6,8 +6,11 @@
+ #include <linux/sched.h>
+ #include "sched.h"
+ 
+-extern void __orig_set_rq_offline(struct rq*);
+-extern void __orig_set_rq_online(struct rq*);
++#define __orig_set_rq_online vmlinux_set_rq_online
++#define __orig_set_rq_offline vmlinux_set_rq_offline
++extern void __mod_set_rq_offline(struct rq*);
++extern void __mod_set_rq_online(struct rq*);
++
+ extern unsigned int process_id[];
+ 
+ extern struct sched_class __orig_stop_sched_class;
+@@ -40,6 +43,37 @@ struct sched_class *mod_class[] = {
+ #define NR_SCHED_CLASS 5
+ struct sched_class bak_class[NR_SCHED_CLASS];
+ 
++static void vmlinux_set_rq_online(struct rq *rq)
++{
++	if (!rq->online) {
++		const struct sched_class *class;
++
++		cpumask_set_cpu(rq->cpu, rq->rd->online);
++		rq->online = 1;
++
++		for_each_class(class) {
++			if (class->rq_online)
++				class->rq_online(rq);
++		}
++	}
++}
++/* DON'T MODIFY INLINE EXTERNAL FUNCTION vmlinux_set_rq_online */
++
++static void vmlinux_set_rq_offline(struct rq *rq)
++{
++	if (rq->online) {
++		const struct sched_class *class;
++
++		for_each_class(class) {
++			if (class->rq_offline)
++				class->rq_offline(rq);
++		}
++
++		cpumask_clear_cpu(rq->cpu, rq->rd->online);
++		rq->online = 0;
++	}
++}
++/* DON'T MODIFY INLINE EXTERNAL FUNCTION vmlinux_set_rq_offline */
+ 
+ static inline void do_write_cr0(unsigned long val)
+ {
+@@ -70,11 +104,11 @@ void clear_sched_state(bool mod)
+ {
+ 	struct task_struct *g, *p;
+ 	struct rq *rq = this_rq();
+-	int queue_flags = DEQUEUE_SAVE | DEQUEUE_MOVE | DEQUEUE_NOCLOCK;
++	int queue_flags = DEQUEUE_SAVE;
+ 
+ 	raw_spin_lock(&rq->lock);
+ 	if (mod) {
+-		set_rq_offline(rq);
++		__mod_set_rq_offline(rq);
+ 	} else {
+ 		__orig_set_rq_offline(rq);
+ 	}
+@@ -86,9 +120,6 @@ void clear_sched_state(bool mod)
+ 		if (p == rq->stop)
+ 			continue;
+ 
+-		/* To avoid SCHED_WARN_ON(rq->clock_update_flags < RQCF_ACT_SKIP) */
+-		rq->clock_update_flags = RQCF_UPDATED;
```

Comment:
I delete these lines previously just for testing. Please make sure if we should keep them.

Reply:
clock_update_flags isn't a member of rq struct.